packages/SwingSet/src/controller.js

```diff
@@ -52,29 +52,38 @@ const KNOWN_CREATION_OPTIONS = harden([
 ]);
 
 /**
- * Scan a directory for files defining the vats to bootstrap for a swingset, and
- * produce a swingset config object for what was found there.  Looks for files
- * with names of the pattern `vat-NAME.js` as well as a file named
- * 'bootstrap.js'.
- *
- * @param basedir  The directory to scan
- *
- * @return a swingset config object: {
- *   bootstrap: "bootstrap",
- *   vats: {
- *     NAME: {
- *       sourceSpec: PATHSTRING
- *     }
- *   }
- * }
- *
- * Where NAME is the name of the vat; `sourceSpec` contains the path to the vat with that name.  Note that
+ * @typedef {Object} SwingSetConfigProperties
+ * @property {string} [sourceSpec] path to the source code
+ * @property {string} [bundleSpec]
+ * @property {Record<string, any>} [parameters]
+ */
+
+/**
+ * @typedef {Record<string, SwingSetConfigProperties>} SwingSetConfigDescriptor
+ * Where the property name is the name of the vat.  Note that
  * the `bootstrap` property names the vat that should be used as the bootstrap vat.  Although a swingset
  * configuration can designate any vat as its bootstrap vat, `loadBasedir` will always look for a file named
  * 'bootstrap.js' and use that (note that if there is no 'bootstrap.js', there will be no bootstrap vat).
+ */
+
+/**
+ * @typedef {Object} SwingSetConfig a swingset config object
+ * @property {string} bootstrap
+ * @property {SwingSetConfigDescriptor} [vats]
+ * @property {SwingSetConfigDescriptor} [bundles]
```

Comment:
Also need devices and (I think) one or two other properties.

Reply:
I did a quick scan, I think adding `@property {any} devices` is probably sufficient.